src/project/projectSlice.ts

```diff
@@ -15,42 +15,100 @@
  * along with this program. If not, see https://www.gnu.org/licenses/.
  */
 
-import { createSlice } from '@reduxjs/toolkit';
+import { createAction, createSlice } from '@reduxjs/toolkit';
+import { setUsers, User } from 'terraso-client-shared/account/accountSlice';
+import {
+  Membership,
+  setMembers,
+} from 'terraso-client-shared/memberships/membershipsSlice';
 import * as projectService from 'terraso-client-shared/project/projectService';
-import { createAsyncThunk } from 'terraso-client-shared/store/utils';
+import { setSites, Site } from 'terraso-client-shared/site/siteSlice';
+import {
+  createAsyncThunk,
+  dehydrated,
+} from 'terraso-client-shared/store/utils';
+
+const { plural: dehydrateProjects, sing: dehydrateProject } = dehydrated<
```

Comment:
Suggestion: It seems to me that the process of changing the structure of the data that comes from the API to something compatible with the client code belongs in the service file not the slice file